src/vs/workbench/services/history/browser/historyService.ts

```diff
@@ -783,6 +783,18 @@ export class HistoryService extends Disposable implements IHistoryService {
 		if (isEditorInput(historyInput)) {
 			this.editorHelper.onEditorDispose(historyInput, () => this.updateHistoryOnEditorDispose(historyInput), this.editorHistoryListeners);
 		}
+
+		// If the original source is a typed editor
+		// and its source should not be included
+		// then remove it on editor dispose
+		else if (isEditorInput(editor) && !this.includeInHistory(historyInput)) {
+			this.editorHelper.onEditorDispose(editor, () => {
+				// Check one more time in case config changes
+				if (!this.includeInHistory(historyInput)) {
```

Comment:
`includeInHistory` will always return true for `EditorInput`, and `editor` here is `EditorInput | undefined`

https://github.com/microsoft/vscode/blob/eab981234f42194a815a62af1aeaa39d450614c8/src/vs/workbench/services/history/browser/historyService.ts#L753

I'm not sure should I change `includeInHistory` function so I just changed this part

If we actually exclude these files from the get go, we can't use quick open to switch to these files while they are still opened which is something you can do to non files (e.g. settings page), so I'm not sure would we want that

Reply:
I am actually having second thoughts about the `includeInHistory` method. I would argue if the user configured an exclude, we should not show the editor in the quick pick history even if it is opened as editor somewhere. Today I do not really remember why I did the change, but I wonder if we should just change it to:

```ts
private includeInHistory(editor: EditorInput | IResourceEditorInput): boolean {
	return !this.resourceExcludeMatcher.value.matches(editor.resource);
}
```

?

Reply:
I like this idea a lot and actually that's what I thought about first, the only reason I didn't do this is because I'm not sure the original intent

We will use ctrl + tab to switch between opened editors anyway, doesn't matter if they're in quick open or not

Reply:
Yeah, there was a time where we did not have the editor quick pick that drives `Ctrl+Tab` and maybe then it was making sense to keep editors that are opened, but not today anymore imho.

However such a change I would make only early in the iteration and now we are at the end. I will also be out in October, so I can only review this in November then.

Reply:
So, should I change it now or maybe till you back in November?

Reply:
Feel free to change it now. I also do not mind that maybe @TylerLeonhardt takes this PR over in October then, since he is not out during that time.

Reply:
OK, thank you very much, have a nice day :)